### ISSUE TITLE
BUG: Iteration index initialisation is wrong in scipy.optimize.linesearch.scalar_search_wolfe2

### ISSUE THREAD

**Issue by Joshuaalbert** [scipy/scipy#12157]
My issue is about the initialisation of the iteration index in the [line search function which produces a step size satisfying the strong Wolfe conditions (`scipy.optimize.linesearch.scalar_search_wolfe2`)](https://github.com/scipy/scipy/blob/v1.4.1/scipy/optimize/linesearch.py#L411).

#### Reproducing code example:
The algorithm comes from page 60 of [Wright and Nocedal, 'Numerical Optimization', 1999](https://pages.mtu.edu/~struther/Courses/OLD/Sp2013/5630/Jorge_Nocedal_Numerical_optimization_267490.pdf).

The index should be initialised a `i=1` but in scipy it is `for i in xrange(...)` which starts at 0. This has an impact on the chosen path the algorithm takes. Specifically, the condition on line 430 sometimes wont be True and an alternative route will be chosen,

``` python
if (phi_a1 > phi0 + c1 * alpha1 * derphi0) or \
           ((phi_a1 >= phi_a0) and (i > 1)):
    ...
```

I noticed this when implementing the line search for JAX where I use scipy's implementation for a unittest. I was inspecting why I woud sometimes get a difference. The purpose of the `i>1` part is so that that statement is true only when at least one iteration has occured. In scipy's implementation, it implies that it's true only when at least two iterations have occured. This might cause extra function evaluations, or simply make the line search fail in worse case. Quite a simple fix though.

#### Scipy/Numpy/Python version information:
```
1.4.1 1.18.4 sys.version_info(major=3, minor=6, micro=10, releaselevel='final', serial=0)
```

**Comment by pedrooa** [comment]
can i work on this?

**Comment by Joshuaalbert** [comment]
@pedrooa let me know if you need any explanation from me

**Comment by st--** [comment]
@pedrooa I've addressed this issue in #13073.
